### PR TITLE
[FLINK-7703] Port JobExceptionsHandler to new REST endpoint

### PR DESCRIPTION
**Pull request by zjureel**
## What is the purpose of the change

Port JobExceptionsHandler to new REST endpoint

## Brief change log
  - *Create JobExceptionsHandler/JobExceptionsInfo/JobExceptionsHeaders class*
  - *Add JobExceptionsHandler to DispatcherRestEndpoint*

## Verifying this change
This change added tests and can be verified as follows:

  - *Add test case JobExceptionsInfoTest for JobExceptionsInfo*
  - *Add test case JobExceptionsInfoNoRootTest for JobExceptionsInfo with no root exception*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+			}
+
+			return new JobExceptionsInfo(jobId, jobName, executionExceptionsInfo);
+		}
+	}
+
+	/**
+	 * Nested class to encapsulate the task execution exception.
+	 */
+	public static final class ExecutionTaskExceptionInfo {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java]: rename to ExecutionExceptionInfo

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	public static final String FIELD_NAME_EXECUTION_EXCEPTIONS = "execution-exceptions";
+
+	private final JobID jobId;
+
+	private final String jobName;
+
+	@Nullable
+	private final ExecutionExceptionsInfo executionExceptionsInfo;
+
+	public JobExceptionsInfo(JobID jobId, String jobName, @Nullable ExecutionExceptionsInfo executionExceptionsInfo) {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java]: This class doesn't appear to resemble the response that the existing JobExceptionsHandler returns. The ExecutionExceptionsInfo contains everything we need.

What is this class for?

**Comment by zjureel** [comment]
@zentol Thank you for your review. I think you're right about ExecutionExceptionsInfo and JobExceptionsInfo, and I have fixed them already. THX

**Comment by zentol** [comment]
merging.

**Comment by zentol** [comment]
@zjureel Can you close this PR? I've already merged it.

**Pull request closed without merging**